code/datums/components/item_two_handed.dm

```diff
@@ -0,0 +1,339 @@
+	parent_item.update_icon()
+
+	if(iscyborg(user))
+		to_chat(user, "<span class='notice'>You dedicate your module to [parent].</span>")
+	else
+		to_chat(user, "<span class='notice'>You grab [parent] with both hands.</span>")
+
+	// Play sound if one is set
+	if(wieldsound)
+		playsound(parent_item.loc, wieldsound, 50, TRUE)
+
+	// Let's reserve the other hand
+	var/obj/item/offhand/offhand_item = new(user)
+	offhand_item.name = "[parent_item.name] - offhand"
+	offhand_item.desc = "Your second grip on [parent_item]."
+	offhand_item.wielded = TRUE
+	user.put_in_inactive_hand(offhand_item)
+
+/**
+ * Unwield the two handed item
+ *
+ * vars:
+ * * user The mob/living/carbon that is unwielding the item
+ * * show_message (option) show a message to chat on unwield
+ */
+/datum/component/two_handed/proc/unwield(mob/living/carbon/user, show_message=TRUE)
+	if(!wielded || !user)
+		return
+
+	// wield update status
+	wielded = FALSE
+	UnregisterSignal(user, COMSIG_MOB_SWAP_HANDS)
+	SEND_SIGNAL(parent, COMSIG_TWOHANDED_UNWIELD, user)
+	if(on_unwield_callback)
+		on_unwield_callback.Invoke(user)
+
+	// update item stats
+	var/obj/item/parent_item = parent
+	if(force_multiplier)
+		parent_item.force /= force_multiplier
+	else if(force_unwielded)
+		parent_item.force = force_unwielded
+
+	// update the items name to remove the wielded status
+	var/sf = findtext(parent_item.name, " (Wielded)", -10) // 10 == length(" (Wielded)")
+	if(sf)
+		parent_item.name = copytext(parent_item.name, 1, sf)
+	else
+		parent_item.name = "[initial(parent_item.name)]"
+
+	// Update icons
+	parent_item.update_icon()
+	if(user.get_item_by_slot(ITEM_SLOT_BACK) == parent)
+		user.update_inv_back()
+	else
+		user.update_inv_hands()
+
+	// Show message if requested
+	if(show_message)
+		if(iscyborg(user))
+			to_chat(user, "<span class='notice'>You free up your module.</span>")
+		else if(require_twohands)
+			to_chat(user, "<span class='notice'>You drop [parent].</span>")
+		else
+			to_chat(user, "<span class='notice'>You are now carrying [parent] with one hand.</span>")
+
+	// Play sound if set
+	if(unwieldsound)
+		playsound(parent_item.loc, unwieldsound, 50, TRUE)
+
+	// Remove the object in the offhand
+	var/obj/item/offhand/offhand_item = user.get_inactive_held_item()
+	if(offhand_item && istype(offhand_item))
+		offhand_item.unwield()
+
+/**
+ * on_attack triggers on attack with the parent item
+ */
+/datum/component/two_handed/proc/on_attack(obj/item/source, mob/living/target, mob/living/user)
+	if(wielded && attacksound)
+		var/obj/item/parent_item = parent
+		playsound(parent_item.loc, attacksound, 50, TRUE)
+
+/**
+ * on_update_icon triggers on call to update parent items icon
+ *
+ * Checked in the order listed, first found is used
+ * Invokes the Icon Update Callback with wield status
+ * Updates the icon using icon_prefix if set
+ */
+/datum/component/two_handed/proc/on_update_icon(datum/source)
+	if(icon_update_callback)
+		icon_update_callback.Invoke(wielded)
```

Comment:
You don't need this, either let the parent handle update_icon normaly or handle it here and block the parent doing anything.

Reply:
Callback was removed.